hbase-client/src/main/java/org/apache/hadoop/hbase/client/ZKConnectionRegistry.java

```diff
@@ -218,7 +236,9 @@ public CompletableFuture<RegionLocations> getMetaRegionLocations() {
     return tracedFuture(() -> {
       CompletableFuture<RegionLocations> future = new CompletableFuture<>();
       addListener(
-        zk.list(znodePaths.baseZNode).thenApply(children -> children.stream()
+        zk.listWithTimeout(znodePaths.baseZNode,
+          conf.getInt(ZK_REGISTRY_ASYNC_GET_TIMEOUT, DEFAULT_ZK_REGISTRY_ASYNC_GET_TIMEOUT),
```

Comment:
Let's store this timeout value as a class field so we do not need to get it everytime?